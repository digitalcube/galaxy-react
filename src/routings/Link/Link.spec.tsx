import React from 'react';
import { render } from '@testing-library/react';
import { Link as RRLink, BrowserRouter } from 'react-router-dom';
import Link from './Link';
import LinkTagProvider, { LinkTagType } from '../../providers/LinkTagProvider';

describe('Link', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Link />);
    expect(baseElement).toBeTruthy();
  });
  it.each([['a' as const], [RRLink]])(
    'link type is %p, should match snapshot',
    (type: LinkTagType) => {
      const { asFragment } = render(
        <BrowserRouter>
          <LinkTagProvider linkType={type}>
            <Link href="/">Home</Link>
            <Link href="https://example.com">External</Link>
          </LinkTagProvider>
        </BrowserRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    }
  );
  it('should not throw error when not using a Link tag of react-router-dom', () => {
    expect(() => {
      render(
        <LinkTagProvider linkType={'a'}>
          <Link href="/">Home</Link>
        </LinkTagProvider>
      );
    }).not.toThrowError();
  });
  it('should throw error when not using a Link tag of react-router-dom', () => {
    expect(() => {
      render(
        <LinkTagProvider linkType={RRLink}>
          <Link href="/">Home</Link>
        </LinkTagProvider>
      );
    }).toThrow(
      'Invariant failed: You should not use <Link> outside a <Router>'
    );
  });
});
