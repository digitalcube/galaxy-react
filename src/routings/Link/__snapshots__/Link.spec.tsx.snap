// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link link type is "a", should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="/"
    to="/"
  >
    Home
  </a>
  <a
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    External
  </a>
</DocumentFragment>
`;

exports[`Link link type is {"$typeof": Symbol(react.forward_ref), "propTypes": [Object], "render": [Function anonymous]}, should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="/"
  >
    Home
  </a>
  <a
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    External
  </a>
</DocumentFragment>
`;
